### ISSUE TITLE
Broken MouseButton events instead of ScreenTouch on Fire HD 8 (10th gen)

### ISSUE THREAD

**Issue by exoticorn** [godotengine/godot#69536]
### Godot version

v4.0-beta7

### System information

Fire HD 8 (10th gen), Android 9, API Level 28, Fire OS 7

### Issue description

When running an android export of a Godot 4 game on my Fire HD 8, the game only receives `InputEventMouseButton` events with `button_index=0`, which is not a valid mouse button. It does not receive any `InputEventScreenTouch` events.

The same export running on my Pixel 6 does receive ScreenTouch events as expected and creates valid emulated MouseButton events (`button_index=1`).

This means that even with "Emulate Mouse From Touch" it is not possible to press any buttons on the Fire tablet.

With Godot 3.5.1 I do receive ScreenTouch events on the Fire tablet.

### Steps to reproduce

I'm including a very simple input test project. It has a single button and prints a message when pressed. It also traces out all input events:

```
func _input(event):
	print(event)
```

I have included versions for both Godot 4 and Godot 3.

Running the Godot 4 version on a Fire HD 8 (10th gen) I observe that I cannot press the button and see no `InputEventScreenTouch` in the output.

Here is the output of me touching the button on my Fire HD 8:

```
InputEventMouseButton: button_index=0, mods=none, pressed=true, position=((661.4832, 401.4981)), button_mask=0, double_click=false
InputEventMouseButton: button_index=0, mods=none, pressed=false, position=((661.4832, 401.4981)), button_mask=0, double_click=false
```
Note the missing "Button pressed" trace and `button_index=0`.

Touching the button on my Pixel 6 works as expected:
```
InputEventMouseButton: button_index=1, mods=none, pressed=true, position=((1215, 535)), button_mask=1, double_click=false
InputEventScreenTouch: index=0, pressed=true, position=((1215, 535)), double_tap=false
InputEventMouseButton: button_index=1, mods=none, pressed=false, position=((1215, 535)), button_mask=0, double_click=false
Button pressed
InputEventScreenTouch: index=0, pressed=false, position=((1215, 535)), double_tap=false
```

 For reference, the Godot 3 version gives the same output on both devices:

```
[InputEventMouseButton:1215]
[InputEventScreenTouch:1217]
[InputEventMouseButton:1221]
Button pressed
[InputEventScreenTouch:1223]
```

### Minimal reproduction project

[InputTest.zip](https://github.com/godotengine/godot/files/10146973/InputTest.zip)

**Comment by exoticorn** [comment]
One more observation: On the Fire HD 8 only the first touch (in a multi-touch situation) generates events.

**Comment by exoticorn** [comment]
By adding traces in `GodotInputHandle.java` I was able to see what kind of events come in from the OS:

On the Fire tablet, I'm getting calls to `handleMotionEvent` with an event source of `0x5002`, ie. both `SOURCE_TOUCHSCREEN` (`0x1002`) and `SOURCE_STYLUS` (`0x4002`).

One simple fix I could see would be to add
```
if ((eventSource & SOURCE_TOUCHSCREEN) == SOURCE_TOUCHSCREEN) {
  return false;
}
```
in `isMouseEvent`.

**Comment by dave2** [comment]
Might dupe: https://github.com/godotengine/godot/issues/69517

**Comment by exoticorn** [comment]
I have implemented my suggested simple fix in https://github.com/exoticorn/godot/tree/fix-android-touch-input and uploaded export templates with this fix to https://share.exoticorn.de/godot4-git-input-fix-android_export_templates.zip

If this can be shown to also fix #69517 I would create a PR from it.
